app/controllers/journeys/income/NonTurnoverIncomeAmountController.scala

```diff
@@ -59,7 +59,10 @@ class NonTurnoverIncomeAmountController @Inject() (override val messagesApi: Mes
         .bindFromRequest()
         .fold(
           formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode, request.userType, taxYear, businessId))),
-          answer => service.persistAnswerAndRedirect(page, businessId, request, answer, taxYear, mode)
+          answer => {
+            val updatedMode = if (request.getValue(page, businessId).contains(answer)) mode else NormalMode
+            service.persistAnswerAndRedirect(page, businessId, request, answer, taxYear, updatedMode)
```

Comment:
Currently, if I fill out Turnover and NonTurnover pages then I get the correct allowance cap on the final page. I can then go back and change either of these amount pages and return to CYA for submission without the allowance being updated. The cap or allowance amount might no longer be viable.

This change means that if the user changes their answer for either of these pages they must go through the rest of the journey, confirming their values with the updated allowance cap